### ISSUE TITLE
Set up AUR package

### ISSUE THREAD

**Issue by gunar** [NebulousLabs/skynet-cli#11]
Setting up skynet-cli as an [AUR](https://aur.archlinux.org/) package should help increase adoption by Arch/Manjaro users.

**Comment by m-cat** [comment]
Thanks for the issue. @ro-tex, do you have experience with this?

Would also be more than happy to accept a PR from the community.

In the meantime, it should be easy to install skynet-cli with Go or build it oneself.

**Comment by ro-tex** [comment]
To answer the question: No, I don't have experience setting up Arch packages.

Regarding the proposal itself, I think that `skynet-cli` is a library for Go developers, regardless of their platform of choice. As such, it's best for it to adhere to the standards of the Go world and be available via `go get` rather than a package manager. Similarly, I would expect from a Rust application or library to be available via `cargo` and a Node.js one to be available via `npm` and `yarn`.

That being said, I might not have the full picture here. @gunar do you think there is a compelling reason for an Arch package to be created and maintained? How would that be more helpful than the regular `go get`? I am not an Arch user, so I might be missing some critical peculiarity of the platform that makes a package reasonable.

**Comment by gunar** [comment]
Hi guys,

Here's the thing. I'm not a golang developer myself. I don't even have go installed on this machine, and I don't know how golang's package manager works—nor rust's for that matter. Is this package intended for golang developers only? I don't think so, right? So adding it to AUR (and apt-get, etc) would allow for any linux user to quickly and easily install your tool and receive updates. I myself haven't installed it because I don't want to think and have to maintain yet another package manager on my machine.

I believe the present CLI tool can accelerate Skynet's adoption—it's free web hosting, who doesn't like that—and by adding it to AUR you remove friction on the onboarding process. That's my 1 cent.

Btw, if you decided to create an AUR package you can tie it to git commits or releases, making for a maintenance-free solution.

**Comment by m-cat** [comment]
Fair enough @gunar. We definitely want the CLI to be on multiple package managers including apt, homebrew etc. They are convenient and the first place many people will check. I don't have much experience with package managers personally so you may have to be patient.

**Comment by gunar** [comment]
No worries! Shameless plug, happy to work with you. It'd take me only a couple of work hours to get it set up on AUR (the one I'm familiar with). Either way, wishing you guys good luck. Love the project.

**Comment by m-cat** [comment]
We're always happy to accept contributions. No pressure though. Even just an outline of what needs to be done would be very appreciated!

**Comment by gunar** [comment]
FWIW I'm doing this for now
```sh
function skynet() {
  curl -X POST "https://siasky.net/skynet/skyfile" -F "file=@$1" \
    | jq ".skylink" | xargs -I _ echo "https://siasky.net/_"
}
```

PS: Congrats on closing the fundraising round!

**Comment by m-cat** [comment]
Seems convenient, not sure what the relevance is to package managers though 🙃

BTW I added a Homebrew recipe and installation instructions to the README

**Comment by gunar** [comment]
That's a fair point. I'd named this issue in terms of a feature but
ultimately my customer need was to be able to use the CLI to upload to
Skynet.

**Comment by m-cat** [comment]
Gotcha, I forgot that you had said you couldn't install the CLI on your machine.

Do you mind if I add that snippet to our docs?

**Comment by gunar** [comment]
Not at all! Be my guest.
